LiteCore/Query/SQLiteFleeceEach.cc

```diff
@@ -214,14 +214,19 @@ class FleeceCursor : public sqlite3_vtab_cursor {
             return SQLITE_OK;
 
         // Parse the Fleece data:
-        slice data = valueAsSlice(argv[0]);
-        if (!data) {
+        slice body = valueAsSlice(argv[0]);
+        if (!body) {
             // Weird not to get a document; have to return early to avoid a crash.
             // Treat this as an empty doc. (See issue #379)
             Warn("fleece_each filter called with null document! Query is likely to fail. (#379)");
             return SQLITE_OK;
         }
-        data = _vtab->context.delegate->fleeceAccessor(data);
+        slice data = _vtab->context.delegate->fleeceAccessor(body);
+        
+        // CBL-1248: if the original data is not document body, getting data
+        // from db's fleece accessor will return nullslice; fallback to use the original data.
+        if (!data.buf)
```

Comment:
There is a better test for this.  If `idxNum` is `kPathIndex` then you should use the `fleeceAccessor`, otherwise do not.  So I think it would be better to simply guard the old line 224 with this condition.  If it is a nested value, then the index will be `kFleeceDataIndex`

Reply:
I'm trying to understand the difference b/w kPathIndex and kFleeceDataIndex. It seems like it's not necessary that when the index is PathIndex, the data always requires database's fleece accessor for the case like `fl_each(fl_nested_value(_V.body, 'sub_dict'), 'array_in_sub_dict')`.